docs/content/latest/develop/learn/ttl-data-expiration-ycql.md

```diff
@@ -108,6 +108,51 @@ SELECT * FROM pageviews;
+
+This feature is available in 2.6.10+, 2.8., and 2.12.1+
+
+#### Configuring for new YCQL time series datasets with TTL
+
+If configuring a new YCQL database for time series datasets and using a default time to live, we recommend the following TServer flag configurations:
+
+##### --tablet_enable_ttl_file_filter = true
+
+Enables expired files to be directly deleted, rather than relying on garbage collection during compaction.
+
+##### --rocksdb_max_file_size_for_compaction = \[the amount of data to be deleted at once, in bytes\]
+
+This value for this flag is chosen depending on how much data is expected to be deleted at once. For example, if a table's TTL is 90 days, it might be desireable to delete three days' worth of data at once. In this case, *rocksdb_max_file_size_for_compaction* should be set to the amount of data expected to be generated in 3 days. Files over this size will be excluded from normal compactions, leading to CPU gains.
+
+Note that there is some tradeoff here between the number of files created and read perforance. A reasonable rule of thumb is to configure the flag such that 30 to 50 files store the full dataset (e.g. 90 days divided by 3 days is 30 files). CPU benefits of using this feature should more than make up for any read performance loss.
+
+##### --sst_files_soft_limit = \[number of expected files at steady state + 20\]
+##### --sst_files_hard_limit = \[number of expected files at steady state + 40\]
+
+The value of these flags is directly dependent on the number of files expected to hold the full dataset at steady state. These flags throttle writes to YCQL if the number of files per tablet exceed their value. Thus, they need to be set in a way that will In the example above, 30 files should hold 90 days worth of data. In this case, *sst_files_soft_limit* would be set to 50, and *sst_files_hard_limit* set to 70.
+
+In some fresh dataset cases, new data will be backfilled into the database before the application is turned on. This backfilled data may have a value-level TTL associated with it that is significantly lower than the `default_time_to_live` property on the table, with the desired effect being that this data be removed earlier than the table TTL would allow. By default, such data will *not* expire early. However, the **file_expiration_value_ttl_overrides_table_ttl** flag can be used to ignore table TTL and expire solely based on value TTL.
+
+**NOTE:** When using the `file_expiration_value_ttl_overrides_table_ttl` flag, be sure to set the flag back to `false` *before* all backfilled data has fully expired. Failing to do so can result in unexpected loss of data. For example, if the `default_time_to_live` is 90 days, and data has been backfilled with value-level TTL from 1 day to 89 days, it is important that the `file_expiration_value_ttl_overrides_table_ttl` flag be set back to `false` within 89 days of data ingestion to avoid data loss.
```

Comment:
{{< warning title="Warning" >}}
When using the `file_expiration_value_ttl_overrides_table_ttl` flag, be sure to set the flag back to `false` *before* all backfilled data has fully expired. Failing to do so can result in unexpected loss of data. For example, if the `default_time_to_live` is 90 days, and data has been backfilled with value-level TTL from 1 day to 89 days, it is important that the `file_expiration_value_ttl_overrides_table_ttl` flag be set back to `false` within 89 days of data ingestion to avoid data loss.
{{< /warning >}}